Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
```

Comment:
current code:
```
UnTrack()
Track()
PyObject_CallFinalizerFromDealloc()
UnTrack()
```

I proposed:

```
PyObject_CallFinalizerFromDealloc()
UnTrack()
```

34fae03 does other things between UnTrack() and Track()
But this code does only `Future_CheckExact()` and it is safe.